datafusion/functions/src/math/gcd.rs

```diff
@@ -142,4 +143,22 @@ mod test {
         assert_eq!(ints.value(2), 5);
         assert_eq!(ints.value(3), 8);
     }
+
+    // from issue https://github.com/apache/datafusion/issues/11031 we know that the previous implementation could
+    // not handle cases were one or both of the inputs were an i64::MAX or i64::MIN coupled with other values (neg or pos)
+    #[test]
+    fn test_gcd_i64_maxes() {
```

Comment:
I think these tests can be implemented as [slt](https://github.com/apache/datafusion/blob/18042fd69138e19613844580408a71a200ea6caa/datafusion/sqllogictest/test_files/scalar.slt#L451).